anillo/middlewares/json.py

```diff
@@ -33,17 +33,23 @@ def wrapper(request, *args, **kwargs):
     return wrapper
 
 
-def wrap_json_body(func):
+def wrap_json_body(func=None, *, preserve_raw_body=False):
     """
     A middleware that parses the body of json requests and
     add it to the request under the `body` attribute (replacing
-    the previous value).
+    the previous value). Can preserve the original value in
+    a new attribute `raw_body` if you give preserve_raw_body=True.
     """
 
+    if func is None:
+        return functools.partial(wrap_json_body, preserve_raw_body=preserve_raw_body)
+
     @functools.wraps(func)
     def wrapper(request, *args, **kwargs):
         ctype, pdict = parse_header(request.headers.get('Content-Type', ''))
         if ctype == "application/json":
+            if preserve_raw_body:
```

Comment:
An other approach (may be better but with breaking changes), is not reemplace body in this middleware and use something like `request.data` instead of `request.body` allowing everyone access to the original body using the standard way.